### ISSUE TITLE
Replace s3_direct_upload with active storage direct uploads?

### ISSUE THREAD

**Issue by waynehoover** [thepracticaldev/dev.to#293]
This commit surprised me:

https://github.com/thepracticaldev/dev.to/commit/40a8476704e3080cadbeb177d4102211f3a7a7fa#diff-8fe8bb54a663b8f12c15fab3893e6d36

Why might I be surprised? Well I am the creator of https://github.com/waynehoover/s3_direct_upload and I haven't touched it in years. I'm happy to see that it is still providing value to people, but I'm curious why it was decided to be used here.

Using this gem you have to pull in jquery and jquery-file-upload ( https://github.com/thepracticaldev/dev.to/blob/master/app/views/videos/new.html.erb#L40 ) which seem to be big prices to pay. Plus Rails 5.2 now supports direct uploads natively through active storage: https://edgeguides.rubyonrails.org/active_storage_overview.html#direct-uploads

But first to upgrade to active_storage we need to upgrade to Rails 5.2, so effectively this is blocked until then. 

Happy you are making use of the gem, but I feel like we can do better. That is maybe I should start maintaining s3_direct_upload if people are still using it, or maybe I should just point people to active_storage direct uploads in the readme.

**Comment by benhalpern** [comment]
Thanks for stopping by our humble repo Wayne.

I definitely recall the day s3 uploads went in as a somewhat panicked approach to getting our video transcoding working properly by moving from some other provider to an AWS approach after the other thing was having issues.

So I'm more than happy to revisit this because it was a total hack job in the first place.

The ActiveStorage story seems like the right direction either way. We'll want to holistically think about the move from Carrierwave.

**Comment by waynehoover** [comment]
I suppose we can move to ActiveStorage in a piecemeal way. We can start with video uploads and work from there.

I can create a new legit issue to move video uploads to active storage, which would be blocked by another story to upgrade to rails 5.2. Sound good?

**Comment by pkfrank** [comment]
Hey @waynehoover, @benhalpern will follow up here shortly with a few additional thoughts.

Thanks again for stopping by and helping out.

**Comment by maestromac** [comment]
not stale

**Comment by maestromac** [comment]
not stale
